app_dart/lib/src/service/luci_build_service.dart

```diff
@@ -17,22 +22,34 @@ import 'buildbucket.dart';
 /// and cancel builds for github repos. It uses [config.luciTryBuilders] to
 /// get the list of available builders.
 class LuciBuildService {
-  LuciBuildService(this.config, this.buildBucketClient, this.serviceAccount);
+  LuciBuildService(this.config, this.buildBucketClient, this.serviceAccount,
+      {GithubChecksUtil githubChecksUtil})
+      : githubChecksUtil = githubChecksUtil ?? const GithubChecksUtil();
 
   BuildBucketClient buildBucketClient;
   Config config;
   ServiceAccountInfo serviceAccount;
+  Logging log;
+  GithubChecksUtil githubChecksUtil;
+
   static const Set<Status> failStatusSet = <Status>{
     Status.canceled,
     Status.failure,
     Status.infraFailure
   };
 
-  /// Returns a map of the BuildBucket builds for a given [repositoryName]
+  /// Sets the appengine [log] used by this class to log debug and error
+  /// messages. This method has to be called before any other method in this
+  /// class.
+  void setLogger(Logging log) {
+    this.log = log;
+  }
```

Comment:
Nit: such logic usually sits in the constructor or a function named `init`.

Reply:
The reason for this is that the logger depends on the AppEngine context which only exists when the handler is running. I couldn't find any other way to pass the logger without major refactorings.